Presentation.Web/app/components/org/user/org-user-delete.controller.ts

```diff
@@ -269,22 +376,63 @@
             };
         }
 
+        private updateGroupSelections() {
+            const selectedContractRights = this.vmContractRoot.rights.filter(x => x.selected);
+            const selectedProjectRights = this.vmProjectRoot.rights.filter(x => x.selected);
+            const selectedSystemRights = this.vmSystemRoot.rights.filter(x => x.selected);
+            const selectedDprRights = this.vmDprRoot.rights.filter(x => x.selected);
+            const selectedOrgRights = this.vmOrgRoot.rights.filter(x => x.selected);
+            const selectedAdminRights = this.vmAdminRoot.rights.filter(x => x.selected);
+
+            if (selectedContractRights.length < this.vmContractRoot.rights.length) {
```

Comment:
these 5 lines are repeated over and over again (along with their source which is a projection of the source vm
Combine tat into a function which takes a root and then:

 - Finds the selected
 - Performs the 5 lines